modules/storages/app/common/storages/adapters/providers/sharepoint/managed_folder_identifier.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Adapters
+    module Providers
+      module Sharepoint
+        class ManagedFolderIdentifier
+          CHARACTER_BLOCKLIST = /[\\<>+?:"|\/]/
+
+          def initialize(project_storage)
+            @project_storage = project_storage
+          end
+
+          def name
+            "#{project.name.gsub(CHARACTER_BLOCKLIST, '_')} (#{project.id})"
```

Comment:
:question: Something that I realize just now: Is this replacement something that's also done that way from the sharepoint side or did we make that up?

Do we need to make sure to avoid naming collisions? E.g. with one project being called `cats > 2kg` and the other called `cats < 2kg`.